ansible-ipi-install/roles/node-prep/templates/install-config.j2

```diff
@@ -20,6 +20,9 @@ platform:
     dnsVIP: {{ dnsvip }}
 {% if (release_version[0]|int > 4) or ((release_version[0]|int == 4) and (release_version[2]|int > 3)) %}
     provisioningNetworkInterface: {{ prov_nic }}
+{% endif %}
+{% if bootstraposimage is defined and bootstraposimage|length %}
```

Comment:
One thing to mention here is that you might want to do a check for 4.4+.  As @rlopez133 mentioned in a previous PR, it's not necessarily guaranteed that 4.3 (and earlier versions) will ignore "extra" config content in install-config.yaml going forward.  Or is this supported in 4.3 as well?

Reply:
I believe this is supported in 4.3 as well (based on this git blame https://github.com/openshift/installer/blame/dd71755327f9e46b38708f28114f56b845d0c083/pkg/tfvars/baremetal/baremetal.go where the parameter was included 7 months ago)

Reply:
Ah, that might explain why 4.3 doesn't seem to honor the image locations in metal3-config.yaml.  Thanks for the heads-up!  I'll give this approach a try.

Reply:
Interesting result trying this out.  4.3 does indeed seem to use `bootstrapOSImage` if you put it in install-config.yaml.  But on the other hand, it did not seem to honor what I used for `clusterOSImage`:

```
# BEGIN ANSIBLE MANAGED BLOCK
    bootstrapOSImage: "http://172.22.0.1/rhcos-43.81.202001142154.0-qemu.x86_64.qcow2.gz?sha256=6dbd3513a824cfe6de157e5d86972c2005b23d71bd8b3a61548f7a1f0a516b68"
    clusterOSImage: "http://172.22.0.1/rhcos-43.81.202001142154.0-openstack.x86_64.qcow2.gz?sha256=a1bda656fa0892f7b936fdc6b6a6086bddaed5dafacedcd7a1e811abb78fe3b0"
# END ANSIBLE MANAGED BLOCK
```

`coreos-downloader` logs from the bootstrap VM show the container trying to download from a slightly different URL, so it would seem the `clusterOSImage` was ignored:

```
curl -g -O http://172.22.0.1/images/rhcos-43.81.202001142154.0-openstack.x86_64.qcow2/rhcos-43.81.202001142154.0-compressed.x86_64.qcow2
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Spee
100  832M  100  832M    0     0   829M      0  0:0-:-- --:--:-- --:--:--  833M
```

Anyhow, `bootstrapOSImage` works, so this PR is looking good!

Reply:
> Interesting result trying this out. 4.3 does indeed seem to use `bootstrapOSImage` if you put it in install-config.yaml. But on the other hand, it did not seem to honor what I used for `clusterOSImage`:
> 
> ```
> # BEGIN ANSIBLE MANAGED BLOCK
>     bootstrapOSImage: "http://172.22.0.1/rhcos-43.81.202001142154.0-qemu.x86_64.qcow2.gz?sha256=6dbd3513a824cfe6de157e5d86972c2005b23d71bd8b3a61548f7a1f0a516b68"
>     clusterOSImage: "http://172.22.0.1/rhcos-43.81.202001142154.0-openstack.x86_64.qcow2.gz?sha256=a1bda656fa0892f7b936fdc6b6a6086bddaed5dafacedcd7a1e811abb78fe3b0"
> # END ANSIBLE MANAGED BLOCK
> ```
> 
> `coreos-downloader` logs from the bootstrap VM show the container trying to download from a slightly different URL, so it would seem the `clusterOSImage` was ignored:
> 
> ```
> curl -g -O http://172.22.0.1/images/rhcos-43.81.202001142154.0-openstack.x86_64.qcow2/rhcos-43.81.202001142154.0-compressed.x86_64.qcow2
>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Spee
> 100  832M  100  832M    0     0   829M      0  0:0-:-- --:--:-- --:--:--  833M
> ```
> 
> Anyhow, `bootstrapOSImage` works, so this PR is looking good!
Looking at the link Edu sent, clusterOSImage isn't listed as an option for 4.3 so we definitely will need to check if its 4.4 or not. @abays thanks for testing out both :D